lib/matplotlib/colors.py

```diff
@@ -362,10 +362,16 @@ def to_rgba_array(c, alpha=None):
     if len(c) == 0:
         return np.zeros((0, 4), float)
     else:
-        if np.iterable(alpha):
-            return np.array([to_rgba(cc, aa) for cc, aa in zip(c, alpha)])
+        lens = {len(cc) if isinstance(cc, (list, tuple)) else -1 for cc in c}
```

Comment:
What if one of them is already a NumPy array? I guess it will just take the slow path, then?

Reply:
Yes it will, but that's quite artificial. Users would have to provide something like
~~~
[np.array(r1, g1, b1),
 np.array(r2, g2, b2),
 ...
]
~~~

@anntzer This should get a comment explaining that all branches do the same and the distinction is for performance optimization.

Reply:
@QuLogic yes, @timhoffm's comment about artificiality applies.
@timhoffm sure.

Reply:
I don't know if I'd consider artificial necessarily. If someone called `to_rgba_array` (perhaps indirectly via a getter on some artist) and then passed individual colours from it, then it could be an array.

Reply:
Fair enough, OTOH adding `np.ndarray` to the tuple of instances checked would complicate the code a bit as you can't necessarily call `len` on a ndarray (it may be zero-dim).  There's actually a test that exercises that case...
For now I guess falling back on the slow path is just fine for that case.